pkg/kubelet/volumemanager/reconciler/reconciler.go

```diff
@@ -251,77 +239,287 @@ func (rc *reconciler) reconciliationLoopFunc() func() {
 					if err != nil &&
 						!nestedpendingoperations.IsAlreadyExists(err) &&
 						!exponentialbackoff.IsExponentialBackoff(err) {
-						// Ignore nestedpendingoperations.IsAlreadyExists and exponentialbackoff.IsExponentialBackoff errors, they are expected.
+						// Ignore nestedpendingoperations.IsAlreadyExists && exponentialbackoff.IsExponentialBackoff errors, they are expected.
 						// Log all other errors.
 						glog.Errorf(
-							"operationExecutor.UnmountDevice failed for volume %q (spec.Name: %q) controllerAttachDetachEnabled: %v with err: %v",
+							"operationExecutor.DetachVolume failed for volume %q (spec.Name: %q) controllerAttachDetachEnabled: %v with err: %v",
 							attachedVolume.VolumeName,
 							attachedVolume.VolumeSpec.Name(),
 							rc.controllerAttachDetachEnabled,
 							err)
 					}
 					if err == nil {
-						glog.Infof("UnmountDevice operation started for volume %q (spec.Name: %q)",
+						glog.Infof("DetachVolume operation started for volume %q (spec.Name: %q)",
 							attachedVolume.VolumeName,
 							attachedVolume.VolumeSpec.Name())
 					}
-				} else {
-					// Volume is attached to node, detach it
-					if rc.controllerAttachDetachEnabled || !attachedVolume.PluginIsAttachable {
-						// Kubelet not responsible for detaching or this volume has a non-attachable volume plugin,
-						// so just remove it to actualStateOfWorld without attach.
-						rc.actualStateOfWorld.MarkVolumeAsDetached(
-							attachedVolume.VolumeName, rc.hostName)
-					} else {
-						// Only detach if kubelet detach is enabled
-						glog.V(12).Infof("Attempting to start DetachVolume for volume %q (spec.Name: %q)",
-							attachedVolume.VolumeName,
-							attachedVolume.VolumeSpec.Name())
-						err := rc.operationExecutor.DetachVolume(
-							attachedVolume.AttachedVolume, false /* verifySafeToDetach */, rc.actualStateOfWorld)
-						if err != nil &&
-							!nestedpendingoperations.IsAlreadyExists(err) &&
-							!exponentialbackoff.IsExponentialBackoff(err) {
-							// Ignore nestedpendingoperations.IsAlreadyExists && exponentialbackoff.IsExponentialBackoff errors, they are expected.
-							// Log all other errors.
-							glog.Errorf(
-								"operationExecutor.DetachVolume failed for volume %q (spec.Name: %q) controllerAttachDetachEnabled: %v with err: %v",
-								attachedVolume.VolumeName,
-								attachedVolume.VolumeSpec.Name(),
-								rc.controllerAttachDetachEnabled,
-								err)
-						}
-						if err == nil {
-							glog.Infof("DetachVolume operation started for volume %q (spec.Name: %q)",
-								attachedVolume.VolumeName,
-								attachedVolume.VolumeSpec.Name())
-						}
-					}
 				}
 			}
 		}
 	}
 }
+
+func (rc *reconciler) reconstructFunc() {
+	glog.V(1).Infof("Start the routine to reconstruct volumes from disk directories %q", rc.kubeletPodsDir)
+	rc.reconstructFromDisk(rc.kubeletPodsDir)
+	rc.timeOfLastReconstruct = time.Now()
```

Comment:
Maybe call this first with a defer so that it is set even if reconstructFromDisk fails with panic